pyproject.toml

```diff
@@ -42,3 +42,10 @@ ignore = [
 
 [tool.ruff.lint.flake8-quotes]
 avoid-escape = false
+
+[tool.ruff.format]
+exclude = [
+    "python/ray/core/src/ray/gcs/",
+    "python/ray/serve/_private/benchmarks/streaming/_grpc/test_server_pb2_grpc.py",
+    "doc/external/",
+]
```

Comment:
This section defines the "final" state for `ruff-format`. That is, after removing the `files` section for `ruff-format` in `.pre-commit-config.yaml` and running `pre-commit run ruff-format --all-files`, those files will be excluded, and the command should succeed.